connectors/byoei.py

```diff
@@ -98,6 +99,7 @@ def __init__(
         self.indexed_document_count = 0
         self.indexed_document_volume = 0
         self.deleted_document_count = 0
+        self._logger = logger_ or logger
```

Comment:
I would keep a single style across the board, so `logger_` or `_logger`

Reply:
`logger_` is to shadow the `logger` in outer space, and `_logger` indicates it's a private variable.